pixeltable/index/embedding_index.py

```diff
@@ -136,7 +136,12 @@ def _validate_embedding_fn(cls, embed_fn: func.Function, name: str, expected_typ
         """Validate the signature"""
         assert isinstance(embed_fn, func.Function)
         sig = embed_fn.signature
-        if len(sig.parameters) != 1 or sig.parameters_by_pos[0].col_type.type_enum != expected_type:
+
+        # The embedding function must be a 1-ary function of the correct type. But it's ok if the function signature
+        # has more than one parameter, as long as it has at most one *required* parameter.
+        if (len(sig.parameters) == 0
+            or len(sig.required_parameters) > 1
```

Comment:
Because it's in theory fine if all the function parameters have defaults, as long as there is at least one parameter.
<!-- Reviewable comment -OBkyziZ-hgjj_ORu8Ef:b-10ajy2 -->
<!-- Sent from Reviewable.io -->